In this project we are going to analyze the dateset of heart disease patients

In [1]:
import numpy as np
import pandas as pd

In [4]:
data = pd.read_csv('heart_disease_dataset.csv')
data

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
0,75,Female,228,119,66,Current,Heavy,1,No,No,Yes,8,119,Yes,Atypical Angina,1
1,48,Male,204,165,62,Current,NaN,5,No,No,No,9,70,Yes,Typical Angina,0
2,53,Male,234,91,67,Never,Heavy,3,Yes,No,Yes,5,196,Yes,Atypical Angina,1
3,69,Female,192,90,72,Current,NaN,4,No,Yes,No,7,107,Yes,Non-anginal Pain,0
4,62,Female,172,163,93,Never,NaN,6,No,Yes,No,2,183,Yes,Asymptomatic,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
995,56,Female,269,111,86,Never,Heavy,5,No,Yes,Yes,10,120,No,Non-anginal Pain,1
996,78,Female,334,145,76,Never,NaN,6,No,No,No,10,196,Yes,Typical Angina,1
997,79,Male,151,179,81,Never,Moderate,4,Yes,No,Yes,8,189,Yes,Asymptomatic,0
998,60,Female,326,151,68,Former,NaN,8,Yes,Yes,No,5,174,Yes,Atypical Angina,1


In [16]:
data.head(10)

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
0,75,Female,228,119,66,Current,Heavy,1,No,No,Yes,8,119,Yes,Atypical Angina,1
1,48,Male,204,165,62,Current,NaN,5,No,No,No,9,70,Yes,Typical Angina,0
2,53,Male,234,91,67,Never,Heavy,3,Yes,No,Yes,5,196,Yes,Atypical Angina,1
3,69,Female,192,90,72,Current,NaN,4,No,Yes,No,7,107,Yes,Non-anginal Pain,0
4,62,Female,172,163,93,Never,NaN,6,No,Yes,No,2,183,Yes,Asymptomatic,0
5,77,Male,309,110,73,Never,NaN,0,No,Yes,Yes,4,122,Yes,Asymptomatic,1
6,64,Female,211,105,86,Former,Heavy,8,Yes,Yes,Yes,2,120,No,Typical Angina,1
7,60,Female,208,148,83,Never,Moderate,4,No,Yes,Yes,2,113,Yes,Asymptomatic,1
8,37,Female,317,137,66,Current,Heavy,3,No,Yes,Yes,5,114,No,Non-anginal Pain,0
9,63,Male,204,141,68,Former,Heavy,8,No,Yes,No,3,107,No,Asymptomatic,1


In [17]:
data.tail(10)

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
990,54,Male,304,112,86,Former,NaN,8,Yes,No,Yes,6,111,No,Asymptomatic,1
991,26,Female,215,100,74,Never,Heavy,7,No,Yes,No,10,135,No,Atypical Angina,0
992,28,Female,220,102,73,Current,Moderate,7,Yes,Yes,Yes,10,102,No,Typical Angina,0
993,70,Female,346,110,62,Never,NaN,1,Yes,Yes,No,8,130,Yes,Asymptomatic,1
994,52,Male,248,159,76,Former,Moderate,9,No,Yes,Yes,2,152,Yes,Asymptomatic,1
995,56,Female,269,111,86,Never,Heavy,5,No,Yes,Yes,10,120,No,Non-anginal Pain,1
996,78,Female,334,145,76,Never,NaN,6,No,No,No,10,196,Yes,Typical Angina,1
997,79,Male,151,179,81,Never,Moderate,4,Yes,No,Yes,8,189,Yes,Asymptomatic,0
998,60,Female,326,151,68,Former,NaN,8,Yes,Yes,No,5,174,Yes,Atypical Angina,1
999,53,Male,226,116,82,Current,NaN,6,No,No,Yes,5,161,Yes,Asymptomatic,1


In [18]:
data.sample(5)

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
763,50,Female,317,148,71,Current,Moderate,1,No,No,Yes,4,116,Yes,Typical Angina,0
51,54,Female,348,129,97,Current,Heavy,4,Yes,No,No,9,187,Yes,Atypical Angina,1
109,37,Female,226,173,86,Current,Moderate,7,No,Yes,No,2,139,No,Asymptomatic,0
303,75,Female,246,142,66,Never,Moderate,9,Yes,No,Yes,7,103,Yes,Non-anginal Pain,1
164,52,Male,339,169,88,Never,NaN,8,Yes,No,No,3,157,No,Atypical Angina,1


🔵 Section 1 — Data Understanding (Q1–Q8)

1. What is the shape of the dataset (rows and columns)?

In [5]:
data.shape

(1000, 16)

Dataset have 1000 rows and 16 columns

2. What are the column names and their data types?

In [7]:
data.columns

Index(['Age', 'Gender', 'Cholesterol', 'Blood Pressure', 'Heart Rate',
       'Smoking', 'Alcohol Intake', 'Exercise Hours', 'Family History',
       'Diabetes', 'Obesity', 'Stress Level', 'Blood Sugar',
       'Exercise Induced Angina', 'Chest Pain Type', 'Heart Disease'],
      dtype='object')

these are all the columns of dataset

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Age                      1000 non-null   int64 
 1   Gender                   1000 non-null   object
 2   Cholesterol              1000 non-null   int64 
 3   Blood Pressure           1000 non-null   int64 
 4   Heart Rate               1000 non-null   int64 
 5   Smoking                  1000 non-null   object
 6   Alcohol Intake           660 non-null    object
 7   Exercise Hours           1000 non-null   int64 
 8   Family History           1000 non-null   object
 9   Diabetes                 1000 non-null   object
 10  Obesity                  1000 non-null   object
 11  Stress Level             1000 non-null   int64 
 12  Blood Sugar              1000 non-null   int64 
 13  Exercise Induced Angina  1000 non-null   object
 14  Chest Pain Type          1000 non-null   

data type of each column is mentioned above.

3. How many numerical and categorical columns are there?

In [11]:
data.dtypes.value_counts()

int64     8
object    8
Name: count, dtype: int64

there are:
    numerical columns = 8,
    categorical cloumns = 8

4. What are the basic statistics (mean, median, min, max) for all numerical columns?

In [12]:
data.describe()

,Age,Cholesterol,Blood Pressure,Heart Rate,Exercise Hours,Stress Level,Blood Sugar,Heart Disease
count,1000.000000,1000.000000,1000.0000,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000
mean,52.293000,249.939000,135.2810,79.204000,4.529000,5.646000,134.941000,0.392000
std,15.727126,57.914673,26.3883,11.486092,2.934241,2.831024,36.699624,0.488441
min,25.000000,150.000000,90.0000,60.000000,0.000000,1.000000,70.000000,0.000000
25%,39.000000,200.000000,112.7500,70.000000,2.000000,3.000000,104.000000,0.000000
50%,52.000000,248.000000,136.0000,79.000000,4.500000,6.000000,135.000000,0.000000
75%,66.000000,299.000000,159.0000,89.000000,7.000000,8.000000,167.000000,1.000000
max,79.000000,349.000000,179.0000,99.000000,9.000000,10.000000,199.000000,1.000000


we can see mean, median,min and max for all numerical column

5. Are there any missing values? If yes, in which columns and how many?

In [13]:
data.isnull().sum()

Age                          0
Gender                       0
Cholesterol                  0
Blood Pressure               0
Heart Rate                   0
Smoking                      0
Alcohol Intake             340
Exercise Hours               0
Family History               0
Diabetes                     0
Obesity                      0
Stress Level                 0
Blood Sugar                  0
Exercise Induced Angina      0
Chest Pain Type              0
Heart Disease                0
dtype: int64

yes, there are 340 missing value in Alcohol Intake column.

6. Are there any duplicate rows?

In [14]:
data.duplicated()

0      False
1      False
2      False
3      False
4      False
       ...  
995    False
996    False
997    False
998    False
999    False
Length: 1000, dtype: bool

No, there are no duplicated data.

7. What is the class balance — how many patients have heart disease vs not?

In [19]:
data.sample(5)

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
779,72,Female,280,139,62,Current,NaN,0,No,Yes,Yes,10,156,Yes,Typical Angina,1
460,48,Female,279,164,74,Current,NaN,7,No,Yes,No,2,119,No,Atypical Angina,0
27,63,Male,273,103,64,Never,NaN,6,Yes,Yes,Yes,9,104,No,Typical Angina,1
627,70,Female,188,154,86,Current,Moderate,2,No,No,Yes,4,109,No,Asymptomatic,0
662,49,Male,212,140,75,Never,Heavy,9,Yes,No,No,9,139,No,Atypical Angina,0


In [20]:
data['Heart Disease'].value_counts()

Heart Disease
0    608
1    392
Name: count, dtype: int64

Heart disease patient = 392

Non Heard disease patient = 608

8. What is the age range of patients in the dataset?

In [26]:
youngest_age = data['Age'].min()
oldest_age = data['Age'].max()
age_range = oldest_age - youngest_age
print(f"Youngest Age = {youngest_age}")
print(f"Oldest Age = {oldest_age}")
print(f"Range of Age = {oldest_age - youngest_age}")

Youngest Age = 25
Oldest Age = 79
Range of Age = 54


🟠 Section 2 — Data Cleaning (Q9–Q16)

9. Handle the missing values in the Alcohol Intake column — what strategy will you use and why?